Add aria-current to navigation link, submenu, page list

carolinan opened the pull request:
<!-- Learn the overall process and best practices for pull requests at https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/repository-management.md#pull-requests. -->

<!-- Gutenberg's license is in the process of updating to be dual-licensed under the GPL and MPL. As part of that transition, all new contributions are dual-licensed. For more information, see: https://github.com/WordPress/gutenberg/blob/trunk/LICENSE.md -->

## Description
<!-- Please describe what you have changed or added -->
Add aria-current to links in the navigation block, if the link is to the current page.

Fixes https://github.com/WordPress/gutenberg/issues/35567

## How has this been tested?
<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, tests ran to see how -->
<!-- your change affects other areas of the code, etc. -->

1. Add a site title, home link, post link, submenu item, and page list inside the navigation block.
2. Turn on the screen reader
3. Navigate to the currently open page in the navigation menu using the Tab key.
4. Confirm that the screen reader reads out that the link is the current page.

You can also view source to confirm that aria-current is added.

## Types of changes
<!-- What types of changes does your code introduce?  -->
<!-- Bug fix (non-breaking change which fixes an issue) -->
<!-- New feature (non-breaking change which adds functionality) -->
<!-- Breaking change (fix or feature that would cause existing functionality to not work as expected) -->
Enhancement

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [x] I've tested my changes with keyboard and screen readers. <!-- Instructions: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/accessibility-testing.md -->
- [ ] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [ ] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/code/native-mobile.md -->

carolinan committed:
Add aria-current to navigation link

carolinan commented on packages/block-library/src/navigation-link/index.php:
```diff
@@ -158,14 +158,16 @@ function render_block_core_navigation_link( $attributes, $content, $block ) {
 		)
 	);
-	$html               = '<li ' . $wrapper_attributes . '>' .
+	$html               = '<li ' . $wrapper_attributes . $aria_current . '>' .
```
@TeBenachi actually, the aria-current should be on the `<a>` not the `<li>`, correct?

TeBenachi replied:
Thank you for working on this.  Yes, aria-current should be on `<a>`.

W3C shows `aria-current=“page”` to be used on `<a>` element. 
https://www.w3.org/TR/wai-aria-practices-1.1/examples/breadcrumb/index.html

carolinan committed:
move aria-current to the <a>

carolinan committed:
Add aria-current to submenu

carolinan committed:
add aria-current to page list

carolinan committed:
add aria-current to site title

carolinan commented on packages/block-library/src/site-title/index.php:
```diff
@@ -21,6 +21,8 @@ function render_block_core_site_title( $attributes ) {
 	$tag_name         = 'h1';
 	$align_class_name = empty( $attributes['textAlign'] ) ? '' : "has-text-align-{$attributes['textAlign']}";
 
+	$aria_current  =  is_home() ? ' aria-current="true"' : '';
```
This needs to be updated to work with static front pages.

carolinan committed:
fix spacing

carolinan committed:
correct aria-current for the site title

Correct aria-current for the site title.
Add aria-current also when the title is a link on a static front page.

carolinan committed:
Add aria-current and rel=home to the home link block

carolinan committed:
fix CS

tellthemachines reviewed (approved):
Thanks for fixing this! Code looks good and it's working well (tested with VoiceOver on Safari) ✅
